packages/samples/headless-react/src/components/pager/pager.class.tsx

```diff
@@ -0,0 +1,79 @@
+import {Component} from 'react';
+import {
+  buildPager,
+  Pager as HeadlessPager,
+  PagerOptions,
+  PagerState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+export class Pager extends Component {
+  private controller: HeadlessPager;
+  public state: PagerState;
+  private unsubscribe: Unsubscribe = () => {};
+
+  constructor(props: {}) {
+    super(props);
+
+    const options: PagerOptions = {numberOfPages: 6};
+    this.controller = buildPager(engine, {options});
+    this.state = this.controller.state;
+  }
+
+  componentDidMount() {
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  private previousPage() {
+    this.controller.previousPage();
+  }
+
+  private nextPage() {
+    this.controller.nextPage();
+  }
+
+  private selectPage(page: number) {
+    this.controller.selectPage(page);
+  }
+
+  private isCurrentPage(page: number) {
+    return this.controller.isCurrentPage(page);
+  }
```

Comment:
Let's remove the methods on lines 36 - 48, and directly use the controller inside the `render` method. This will reduce the need to controller and shorten the sample by ~20 lines.